table/column.go

```diff
@@ -697,6 +697,24 @@ func FillVirtualColumnValue(virtualRetTypes []*types.FieldType, virtualColumnInd
 			if err != nil {
 				return err
 			}
+
+			if mysql.HasUnsignedFlag(colInfos[idx].FieldType.GetFlag()) && castDatum.GetUint64() > 0 && !sctx.GetSessionVars().StmtCtx.ShouldClipToZero() {
```

Comment:
What if decimal or float is used and `GetUin64()` would return 0, due to truncation, like if castDatum is float(0.1)?

Reply:
Please add a test where castDatum becomes 0.1 (and I think you need to use `castDatum.GetFloat64() > float64(0)` instead of integer types here.
Otherwise LTGM

Reply:
@mjonss how about remove "castDatum.GetUint64() > 0", if mysql.HasUnsignedFlag(colInfos[idx].FieldType.GetFlag()) && !sctx.GetSessionVars().StmtCtx.ShouldClipToZero(), then check the value?